### PR TITLE
fix: reintroduce gaia docker builds

### LINKED ISSUE #3124: [feature request]: gaia container image is outdated since 2 years
gaia container image is outdated since 2 years (see https://github.com/orgs/cosmos/packages?repo_name=gaia). Reason for this is that the `docker-push` pipeline never runs at the moment, which I assume is due to an incorrect branch check at here: https://github.com/cosmos/gaia/blob/3462c51d6874a82dbd7fe6b295d804e1ffe5caa5/.github/workflows/docker-push.yml#L6

### PR DESCRIPTION
**Pull request by fastfadingviolets**
## Description

Closes: #3124 

Brings back the docker-push workflow to build a docker image for gaia. Gets rid of the `Dockerfile`, and uses only the `e2e.Dockerfile.` Also changes the UID/GUID of the user in the docker image to be `1025`, which makes it compatible with [interchaintest](https://github.com/strangelove-ventures/interchaintest/)

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

* [x] Included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
* [x] Added `!` to the type prefix if API, client, or state breaking change (i.e., requires minor or major version bump)
* [x] Targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
* [x] Provided a link to the relevant issue or specification
* [ ] Followed the guidelines for [building SDK modules](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules)
* [ ] Included the necessary unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
* [x] Added a changelog entry in `.changelog` (for details, see [contributing guidelines](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#changelog))
* [ ] Included comments for [documenting Go code](https://blog.golang.org/godoc)
* [ ] Updated the relevant documentation or specification
* [ ] Reviewed "Files changed" and left comments if necessary 
* [x] Confirmed all CI checks have passed

### CODE REVIEW

**Comment by fastfadingviolets** [comment]
On the `hyphacoop` fork, the image [builds but doesn't push](https://github.com/hyphacoop/gaia/actions/runs/9876360057/job/27275249538) over permissions issues; I don't think this'd be a problem if this were merged.

File: .github/workflows/docker-push.yml [hunk 1]
```diff
@@ -3,7 +3,10 @@ name: Create and publish a Docker image
 
 on:
   push:
-    branches: ['release']
+    branches: ['main']
```
Comment (MSalopek) [inline .github/workflows/docker-push.yml]: Is it possible to do this nigthly and not on every push to main?
Main gets a lot of pushes.

File: .github/workflows/docker-push.yml [hunk 2]
```diff
@@ -33,19 +49,12 @@ jobs:
         with:
           images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
 
-      - name: Build and push Docker image
+      - name: Build and push docker image
         uses: docker/build-push-action@v6.1.0
         with:
           context: .
+          file: e2e.Dockerfile
```
Comment (MSalopek) [inline .github/workflows/docker-push.yml]: Can we rename e2e.Dockerfile to Dockerfile so it's not confusing?

We will be using that file for e2e and providing docker images.

**Review by MSalopek — APPROVED** [review]
Thank you for your contribution!

I think we can merge this, I would like us to switch to using a file named Dockerfile and changing the `docker-push` workflow fto a nightly build - main is quite noisy and gets multiple pushes a day which are not all required to produce working binaries at all times.

Something like [this](https://github.com/n8n-io/n8n/blob/master/.github/workflows/docker-images-nightly.yml#L6), maybe? We can have a separate workflow for nightly and release builds, it does not make much of a difference.

EDIT:
Suggestions are nice to haves, they are not merge blocking.

**Comment by fastfadingviolets** [comment]
Thanks for your comments! I addressed them, plus added a short `DOCKER_README` explaining how to use the image, and how tagging works for it

**Comment by MSalopek** [comment]
> Thanks for your comments! I addressed them, plus added a short `DOCKER_README` explaining how to use the image, and how tagging works for it

Awesome! Thank you very much for this!

**Review by stana-miric — APPROVED** [review]

**Review by MSalopek — APPROVED** [review]
LGTM!

Thanks for the changes!

**Pull request merged**